inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -622,10 +625,58 @@ def load_from_source_data(obj, eng):
     """Restore the workflow data and extra_data from source_data."""
     try:
         source_data = obj.extra_data['source_data']
-        source_data['extra_data']['_task_history'] = obj.extra_data.get('_task_history', [])
+        source_data['extra_data']['_task_history'] = obj.extra_data.get(
+            '_task_history', [])
         obj.data = source_data['data']
         obj.extra_data = source_data['extra_data']
         obj.extra_data['source_data'] = deepcopy(source_data)
         obj.save()
     except KeyError:
-        raise ValueError("Can't start/restart workflow as 'source_data' is either missing or corrupted")
+        raise ValueError("Can't start/restart workflow as"
+                         "'source_data' is either missing or corrupted")
+
+
+def restart_workflow(obj, original_workflow, position=[0]):
+    """Restarts workflow
+
+    Args:
+        wf: Workflow to restart
+        original_workflow: Workflow which restarts
+        position: To which position wf should be restarted
+    """
+    obj.callback_pos = position
+    obj.extra_data['source_data']['extra_data']['delay'] = 10
+    obj.extra_data['source_data']['extra_data'].setdefault(
+        'restarted-by-wf', []).append(original_workflow.id)
+    obj.save()
+    db.session.commit()
+    obj.continue_workflow('restart_task', delayed=True)
+
+
+@with_debug_logging
+def run_next_if_necessary(obj, eng):
```

Comment:
Why not `restart_blocked_workflow`? :)

Reply:
because it's not only restarting, its checking if restart if necessary, then it finds the proper workflow to restart and then restarts it.